### PR TITLE
Added dropdown menu for focus problems.

### PR DESCRIPTION
**Pull request by shaubby**
…oblem in the USACO ide

_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

This will allow users to open focus problems in the USACO ide.

### CODE REVIEW

**Review by bqi343 — CHANGES_REQUESTED** [review]
good suggestion. can you refactor the code so that the functionality common to `ProblemsListItem` and `FocusProblem` isn't duplicated? Same for `ProblemsListItemDropdown` and `FocusProblemDropdown`.

**Comment by shaubby** [comment]
> 

The FocusProblem and ProblemsListItem components were already different components, as they have very different functionalities. However, I can try to integrate FocusProblemDropdown into the prexisting ProblemsListItemDropdown class.

**Comment by shaubby** [comment]
Also, I see that my code had failed one of the tests, do you have any idea what happened?

**Comment by bqi343** [comment]
> The FocusProblem and ProblemsListItem components were already different components, as they have very different functionalities.

I was referring to `more`, which should be the same for both components.

> do you have any idea what happened?

idk

**Comment by shaubby** [comment]
Oh ok, they are slightly different, but I can probably still integrate them both into the ProblemsListItemDropdown component. Also, I was planning on applying to join the CPI web development team, should I finish this and link to it in my application afterwards or just apply now?

**Comment by mjh316** [comment]
> Also, I was planning on applying to join the CPI web development team, should I finish this and link to it in my application afterwards or just apply now?

Awesome to see that you're planning on applying! I think it'd be good to finish it first! :)

**Review by shaubby — COMMENTED** [review]
Cleaned up code and combined files as asked

File: src/components/markdown/FocusProblem.tsx [hunk 1]
```diff
@@ -22,9 +25,13 @@ export default function FocusProblem({
       `The focus problem list ${problemID} should have exactly one problem.`
     );
   }
+  const darkMode = useDarkMode();
```
Comment (bqi343) [inline src/components/markdown/FocusProblem.tsx:28]: what's this for?

File: src/components/markdown/ProblemsList/ProblemsListItemDropdown.tsx [hunk 2]
```diff
@@ -1,22 +1,30 @@
...
 
 export default function ProblemsListItemDropdown(
-  props: ProblemsListItemProps & { onViewProblemSolutions: () => void }
+  props: ProblemsListItemProps & { isFocusProblem: boolean }
 ) {
   const [copied, setCopied] = React.useState(false);
 
   const problem = props.problem;
+  const isFocusProblem = props.isFocusProblem;
+  console.log(isFocusProblem);
```
Comment (bqi343) [inline src/components/markdown/ProblemsList/ProblemsListItemDropdown.tsx:19]: remove

File: src/components/markdown/FocusProblem.tsx [hunk 3]
```diff
@@ -22,9 +25,13 @@ export default function FocusProblem({
...
 
   const problem: ProblemInfo = problemList.problems[0];
 
+  const tippyRef = React.useRef<Instance>();
+  const [isDropdownShown, setIsDropdownShown] = React.useState(false);
```
Comment (bqi343) [inline src/components/markdown/FocusProblem.tsx:33]: these also don't look like they're used

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: src/components/markdown/ProblemsList/ProblemsListItem.tsx [hunk 4]
```diff
@@ -167,52 +165,6 @@ export default function ProblemsListItem(
   const tippyRef = React.useRef<Instance>();
```
Comment (bqi343) [inline src/components/markdown/ProblemsList/ProblemsListItem.tsx:165]: also I think these variables are no longer needed?

**Comment by shaubby** [comment]
Alright, those variables had been used previously but are no longer used now, I just forgot to get rid of them. It should be good to go now.

**Comment by thecodingwizard** [comment]
image

I think there's some unwanted padding here

Might be able to be fixed with

```js
const StyledTippy = styled(Tippy)`
  .tippy-content {
    padding: 0;
  }
`;
```

File: src/components/markdown/ProblemsList/ProblemsListItemDropdown.tsx [hunk 5]
```diff
@@ -40,48 +47,96 @@ export default function ProblemsListItemDropdown(
...
-              window.location.href.split(/[?#]/)[0] +
-                '#problem-' +
-                problem.uniqueId
-            );
-          }}
+    <Tippy
+      onCreate={tippy => (tippyRef.current = tippy)}
+      content={
+        isDropdownShown ? (
+          <div className="-mx-2 text-left">
```
Comment (shaubby) [inline src/components/markdown/ProblemsList/ProblemsListItemDropdown.tsx:77]: It was actually the -mx-2 that fixed it.

**Pull request merged**